pysam/libchtslib.pyx

```diff
@@ -222,6 +226,8 @@ cdef class HTSFile(object):
                 fd = self.filename
             else:
                 fd = self.filename.fileno()
+                
+            dup_fd = dup(fd)
```

Comment:
Should we support passing a `close_file=True` parameter to `HTSFile` objects to control whether the `fd` should be `dup`ed?